src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs

```diff
@@ -50,5 +57,32 @@ public override bool IsLazyEvalParam(int index)
         {
             return LamdaParamMask.TestBit(index);
         }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            // Check if all record names of args exist against arg types and if its possible to coerce.
+            if (isValid)
+            {
+                for (var i = 0; i < args.Length; i++)
+                {
+                    DType curType = argTypes[i];
+
+                    if (curType.IsRecord && !curType.CheckAggregateNames(this.ParamTypes[i], args[i], errors, SupportsParamCoercion, context.Features.PowerFxV1CompatibilityRules))
```

Comment:
>IsRecord [](http://example.com/codeflow?start=32&length=8)

Do we need to do a similar check for table parameters? For example, this check validates that {a:1,b:2,c:3} can't be sent to an argument that expects ![a:n, b:n]. But if we have an argument of type *[a:n, b:n] we probably don't want to allow [{a:1,b:2,c:3}] either

Reply:
updated to check table as well.